header-validator/src/validate-json.ts

```diff
@@ -29,8 +31,10 @@ class Context extends context.Context {
   }
 }
 
+type FieldFunc<I, T, V> = (ctx: Context, i: I, t: Readonly<Partial<T>>) => V
```

Comment:
[optional idea] Adding this partial `t` everywhere adds a non negligible amount of complexity for the use case it solves.

We're already carrying context. Would it make sense to explicitly set what we need on it (startTime & expiry) and use it directly ?

It's certainly less elegant but might be simpler. We could also enforce the order, i.e., throw an error if we get to a point where we expect the value to be set and it is not.